nitrocli/src/commands.rs

```diff
@@ -171,41 +174,30 @@ where
   .map_err(|(_data, err)| err)
 }
 
-/// Pretty print the common status of all Nitrokey devices.
-fn print_status(
-  model: &'static str,
-  smartcard_id: &str,
-  firmware_major: i32,
-  firmware_minor: i32,
-  user_retries: u8,
-  admin_retries: u8,
-) {
+/// Query and pretty print the common status of all Nitrokey devices.
+fn print_status(model: &'static str, device: &nitrokey::DeviceWrapper) -> Result<()> {
+  let serial_number = device
+    .get_serial_number()
+    .map_err(|err| get_error("Could not query the serial number", &err))?;
   println!(
     r#"Status:
   model:             {model}
-  smart card ID:     {id}
+  smart card ID:     0x{id}
```

Comment:
Nit: Can this addition be moved into the previous commit where you factored out this function? Or am I missing something?

Reply:
No.  This change is needed as we replace the value from the `StorageStatus` struct (with leading `0x`) with the value from `Device::get_serial_number` (no leading `0x`) in this commit.